/**:
  ros__parameters:
    namespace_str: "/opp_racecar"
    waypoints_path: "src/pure_pursuit/racelines/adjusted_smoothed_path.csv"
    odom_topic: "/opp_racecar/odom"
    car_refFrame: "opp_racecar/base_link"
    global_refFrame: "map"
    drive_topic: "/opp_drive"
    rviz_current_waypoint_topic: "/opp_racecar/current_waypoint"
    rviz_lookahead_waypoint_topic: "opp_racecar/lookahead_waypoint"

    # Pure Pursuit Parameters
    # Lookahead is computed using the following: lookahead = min(max(min_lookahead, max_lookahead * curr_velocity / lookahead_ratio), max_lookahead);
    # min_lookahead: 0.5
    # max_lookahead: 3.0 # lookahead when the car is going around max velocity
    # lookahead_ratio: 8.0 # approximately the max velocity
    min_lookahead: 0.2
    max_lookahead: 1.0 # lookahead when the car is going around max velocity
    lookahead_ratio: 4.0 # approximately the max velocity
    K_p: 0.5
    steering_limit: 25.0
    velocity_percentage: 1.0 # the limit, can only do 1 lap with clean tires
    